server/src/main/java/org/apache/iotdb/db/engine/compaction/cross/rewrite/RewriteCrossSpaceCompactionSelector.java

```diff
@@ -56,6 +75,223 @@ public RewriteCrossSpaceCompactionSelector(
     this.dataRegionId = dataRegionId;
     this.timePartition = timePartition;
     this.tsFileManager = tsFileManager;
+    this.memoryBudget = config.getCrossCompactionMemoryBudget();
```

Comment:
apply this memory from the storage engine, not from the whole system

Reply:
reslove